modules/decima-model/src/main/java/com/shade/decima/model/rtti/types/jolt/physics/constraints/HingeConstraintSettings.java

```diff
@@ -0,0 +1,37 @@
+package com.shade.decima.model.rtti.types.jolt.physics.constraints;
+
+import com.shade.decima.model.rtti.types.jolt.JoltUtils;
+import com.shade.util.NotNull;
+import org.joml.Vector3f;
+
+import java.nio.ByteBuffer;
+
+public class HingeConstraintSettings extends TwoBodyConstraintSettings {
+
+    Vector3f point1 = new Vector3f(0.0f);
+    Vector3f hingeAxis1 = new Vector3f(0.0f, 1.0f, 0.0f);
+    Vector3f normalAxis1 = new Vector3f(1.0f, 0.0f, 0.0f);
+    Vector3f point2 = new Vector3f(0.0f);
+    Vector3f hingeAxis2 = new Vector3f(0.0f, 1.0f, 0.0f);
+    Vector3f normalAxis2 = new Vector3f(1.0f, 0.0f, 0.0f);
+    float limitsMin = (float) -Math.PI;
+    float limitsMax = (float) Math.PI;
+    float maxFrictionForce = 0.0f;
+    MotorSettings motorSettings = new MotorSettings();
```

Comment:
Make all `private` for now; remove default values as they are immediately overwritten in `restoreBinaryState`